### PR TITLE
Use Lucene provided Persian stem

### LINKED ISSUE #14846: [Feature Request] Add Persian stem support
### Is your feature request related to a problem? Please describe

OpenSearch lacks effective separation of suffixes in Persian due to not use of a stemmer

### Describe the solution you'd like

Using a stemmer for Persian

### Related component

Search

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ebraminio**
### Description
Lucene provided Persian stem apparently isn't hooked yet and this change is doing that based on what is done for Arabic stem support.

### Related Issues
Resolves #14846

### Check List
- [x] Functionality includes testing.
- [x] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
Thanks for this! 

Could you please also take a look at doc and API reference changes needed? 

For the API spec, https://github.com/opensearch-project/opensearch-api-specification likely doesn't have filter examples at all, so may be a little bit of work to add the first one.

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Add Plugin interface for loading application based configuration templates (([#14659](https://github.com/opensearch-project/OpenSearch/issues/14659)))
 - Refactor remote-routing-table service inline with remote state interfaces([#14668](https://github.com/opensearch-project/OpenSearch/pull/14668))
 - Add prefix mode verification setting for repository verification (([#14790](https://github.com/opensearch-project/OpenSearch/pull/14790)))
+- Use Lucene provided Persian stem support (([#14847](https://github.com/opensearch-project/OpenSearch/pull/14847)))
```
Comment (dblock) [inline CHANGELOG.md:24]: I would just say "Added persian_stem filter" or something like that that a user will understand.

**Review by dblock — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ebraminio) [reply CHANGELOG.md:24]: Done

**Comment by ebraminio** [comment]
> For the API spec, https://github.com/opensearch-project/opensearch-api-specification likely doesn't have filter examples at all, so may be a little bit of work to add the first one.

Yes, I couldn't find anything about the Arabic one for example.

**Comment by dblock** [comment]
> > For the API spec, https://github.com/opensearch-project/opensearch-api-specification likely doesn't have filter examples at all, so may be a little bit of work to add the first one.
> 
> Yes, I couldn't find anything about the Arabic one for example.

Would really appreciate any time you can spare!

**Comment by ebraminio** [comment]
> Would really appreciate any time you can spare!

Sure, luckily I could find `PorterStemTokenFilter` for `PorterStemTokenFilterFactory` class so I made based on it https://github.com/opensearch-project/opensearch-api-specification/pull/444

**Review by dblock — APPROVED** [review]

**Comment by ebraminio** [comment]
> The process '/usr/bin/git' failed with exit code 128

Git didn't exists, may I go for a manual back port?

**Comment by dblock** [comment]
@ebraminio Yes please, I don't know what's going on here.

**Pull request merged**